test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java

```diff
@@ -265,6 +272,241 @@ public void testGetDistanceBetweenTwoSCwithStartSCAsUnboundedAndStreamTruncateda
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE + "-1", controller, connectionFactory);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM + "-1", new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        // write events to stream
+        write30ByteEvents(10, writer);
+        Segment segment = new Segment(SCOPE + "-1", STREAM + "-1", 0);
+        StreamCut streamCut = new StreamCutImpl(Stream.of(SCOPE + "-1", STREAM + "-1"),
+                ImmutableMap.of(segment, 270L));
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(SCOPE + "-1", clientConfig);
+        log.info("Done creating batch client factory");
+
+        StreamCut  nextStreamCut = batchClient.getNextStreamCut(streamCut, 93L);
+        assertTrue(nextStreamCut != null);
+        assertTrue(nextStreamCut.asImpl().getPositions().size() == 1);
+        assertTrue(nextStreamCut.asImpl().getPositions().get(segment).equals(300L));
+    }
+
+    @Test(timeout = 50000)
+    public void testNextStreamCut_ScaleUp() throws ExecutionException, InterruptedException {
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.byEventRate(1, 2, 1))
+                .build();
+        Controller controller = controllerWrapper.getController();
+        controllerWrapper.getControllerService().createScope(SCOPE + "-2", 0L).get();
+        controller.createStream(SCOPE + "-2", STREAM + "-2", config).get();
+        @Cleanup
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE + "-2", controller, connectionFactory);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM + "-2", new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+        // write events to stream
+        write30ByteEvents(4, writer);
+
+        Map<Double, Double> keyRanges = new HashMap<>();
+        keyRanges.put(0.0, 0.5);
+        keyRanges.put(0.5, 1.0);
+        Stream stream = new StreamImpl(SCOPE + "-2", STREAM + "-2");
+        // Stream scaling up
+        Boolean status = controller.scaleStream(stream,
+                Collections.singletonList(0L),
+                keyRanges,
+                executorService()).getFuture().get();
+        assertTrue(status);
+
+        write30ByteEvents(6, writer);
+
+        Segment segment0 = new Segment(SCOPE + "-2", STREAM + "-2", 0);
+        Segment segment1 = Segment.fromScopedName(SCOPE + "-2/" + STREAM + "-2/1.#epoch.1");
+        Segment segment2 = Segment.fromScopedName(SCOPE + "-2/" + STREAM + "-2/2.#epoch.1");
+        StreamCut streamCut = new StreamCutImpl(Stream.of(SCOPE + "-2", STREAM + "-2"),
+                ImmutableMap.of(segment0, 120L));
```

Comment:
Here startStreamcut tested is 4*30 = 120 . this is total lenth of segemnt 0 .
One more use case to  be tested when StartStreamCut is less then total length of segment 0 like 60 or 90 .

Reply:
done.